src/lib/parsers/sol-json.c

```diff
@@ -690,6 +690,24 @@ sol_json_serialize_boolean(struct sol_buffer *buffer, bool val)
     return 0;
 }
 
+static int
+_hex_decode(const char *p, uint8_t *val)
+{
+    int16_t n;
+    char hexdigits[3], *endptr;
+
+    hexdigits[2] = '\0';
+    memcpy(hexdigits, p, 2);
+    errno = 0;
+    n = strtoul(hexdigits, &endptr, 16);
```

Comment:
Maybe this is faster/clearer?

      bool
      parse_hex_digit(uint8_t digit, uint8_t *nibble)
      {
        if (digit >= '0' && digit <= '9') {
          *nibble = digit - '0';
          return true;
        }
        if (digit >= 'a' && digit <= 'f') {
          *nibble = digit - 'a';
          return true;
        }
        if (digit >= 'A' && digit <= 'F') {
          *nibble = digit - 'A';
          return true;
        }

        return false;
      }
   
      int
      parse_hex(char *in, uint8_t *out)
      {
        uint8_t nibble;

        if (!parse_hex_digit(in[0], &nibble))
          return -EINVAL;

        *out = nibble;

        if (!parse_hex_digit(in[1], &nibble))
          return -EINVAL;

        *out = (*out << 4) | nibble;

        return 0;
      }